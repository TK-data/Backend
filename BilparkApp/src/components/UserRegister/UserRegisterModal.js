import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavigationActions } from 'react-navigation';
import { Text, Modal, StyleSheet, View, Button } from 'react-native';
import { registerUserModalVisible } from '../../actions/registerUser';


const styles = StyleSheet.create({
  modal: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    padding: 10,
    backgroundColor: '#002776',
  },
  modalText: {
    color: '#fff',
  },
});

const UserRegisterModal = ({ modalVisible, modalTransparent, visibleModal, navigateToLogin }) => {
  return (
    <Modal
      visible={modalVisible}
      animationType="slide"
      transparent={modalTransparent}
      onRequestClose={() => {
        alert('Modal has been closed.');
      }}
    >
      <View style={styles.modal}>
        <Text style={styles.modalText}>Registrering godkjent</Text>
        <Button
          title="GÃ¥ til innlogging"
          onPress={() => {
            visibleModal(false);
            navigateToLogin();
          }}
        />
      </View>
    </Modal>
  );
};

UserRegisterModal.propTypes = {
  modalVisible: PropTypes.bool.isRequired,
  modalTransparent: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => {
  return {
    modalVisible: state.registerUserModalVisible,
    modalTransparent: state.registerUserModalTransparent,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    visibleModal: bool => dispatch(registerUserModalVisible(bool)),
    navigateToLogin: () => dispatch(NavigationActions.back()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserRegisterModal);
