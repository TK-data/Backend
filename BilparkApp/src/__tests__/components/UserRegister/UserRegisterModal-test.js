import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import UserRegisterModal from '../../../components/UserRegister/UserRegisterModal';

const initialState = {
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing UserRegisterModal', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <UserRegisterModal />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
