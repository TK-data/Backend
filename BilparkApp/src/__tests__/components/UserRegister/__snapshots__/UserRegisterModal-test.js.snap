// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserRegisterModal Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserRegisterModal
    modalTransparent={undefined}
    modalVisible={undefined}
    navigateToLogin={[Function]}
    visibleModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "animationType": "slide",
      "children": <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#002776",
            "flex": 1,
            "justifyContent": "center",
            "padding": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#fff",
            }
          }
        >
          Registrering godkjent
        </Text>
        <Button
          onPress={[Function]}
          title="Gå til innlogging"
        />
      </View>,
      "hardwareAccelerated": false,
      "onRequestClose": [Function],
      "transparent": undefined,
      "visible": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Registrering godkjent
          </Text>,
          <Button
            onPress={[Function]}
            title="Gå til innlogging"
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#002776",
          "flex": 1,
          "justifyContent": "center",
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Registrering godkjent",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "#fff",
            },
          },
          "ref": null,
          "rendered": "Registrering godkjent",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onPress": [Function],
            "title": "Gå til innlogging",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "animationType": "slide",
        "children": <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#002776",
              "flex": 1,
              "justifyContent": "center",
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#fff",
              }
            }
          >
            Registrering godkjent
          </Text>
          <Button
            onPress={[Function]}
            title="Gå til innlogging"
          />
        </View>,
        "hardwareAccelerated": false,
        "onRequestClose": [Function],
        "transparent": undefined,
        "visible": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#fff",
                }
              }
            >
              Registrering godkjent
            </Text>,
            <Button
              onPress={[Function]}
              title="Gå til innlogging"
            />,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "#002776",
            "flex": 1,
            "justifyContent": "center",
            "padding": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Registrering godkjent",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#fff",
              },
            },
            "ref": null,
            "rendered": "Registrering godkjent",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onPress": [Function],
              "title": "Gå til innlogging",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
